src/luascript.cpp

```diff
@@ -12245,7 +12256,7 @@ int LuaScriptInterface::luaCombatSetParameter(lua_State* L)
 int LuaScriptInterface::luaCombatSetFormula(lua_State* L)
 {
 	// combat:setFormula(type, mina, minb, maxa, maxb)
-	Combat* combat = getUserdata<Combat>(L, 1);
+	const Combat_ptr combat = getSharedPtr<Combat>(L, 1);
 	if (!combat) {
 		lua_pushnil(L);
```

Comment:
Add this when !combat is true reportErrorFunc(L, getErrorDesc(LUA_ERROR_COMBAT_NOT_FOUND));